kustomize/components/shopping-assistant/scripts/1_deploy_alloydb_infra.sh

```diff
@@ -101,14 +101,15 @@ sed -i "s/ALLOYDB_PRODUCTS_DATABASE_NAME_VAL/${ALLOYDB_PRODUCTS_DATABASE_NAME}/g
 sed -i "s/ALLOYDB_PRODUCTS_TABLE_NAME_VAL/${ALLOYDB_PRODUCTS_TABLE_NAME}/g" kustomize/components/alloydb/kustomization.yaml
 sed -i "s/ALLOYDB_SECRET_NAME_VAL/${ALLOYDB_SECRET_NAME}/g" kustomize/components/alloydb/kustomization.yaml
 
-# Substitute environment values (kubernetes-manifests/shoppingassistantservice.yaml)
-sed -i "s/PROJECT_ID_VAL/${PROJECT_ID}/g" kubernetes-manifests/shoppingassistantservice.yaml
-sed -i "s/REGION_VAL/${REGION}/g" kubernetes-manifests/shoppingassistantservice.yaml
-sed -i "s/ALLOYDB_CLUSTER_NAME_VAL/${ALLOYDB_CLUSTER_NAME}/g" kubernetes-manifests/shoppingassistantservice.yaml
-sed -i "s/ALLOYDB_INSTANCE_NAME_VAL/${ALLOYDB_INSTANCE_NAME}/g" kubernetes-manifests/shoppingassistantservice.yaml
-sed -i "s/ALLOYDB_DATABASE_NAME_VAL/${ALLOYDB_PRODUCTS_DATABASE_NAME}/g" kubernetes-manifests/shoppingassistantservice.yaml
-sed -i "s/ALLOYDB_TABLE_NAME_VAL/${ALLOYDB_PRODUCTS_TABLE_NAME}/g" kubernetes-manifests/shoppingassistantservice.yaml
-sed -i "s/ALLOYDB_SECRET_NAME_VAL/${ALLOYDB_SECRET_NAME}/g" kubernetes-manifests/shoppingassistantservice.yaml
+# Substitute environment values (kustomize/components/shopping-assistant/shoppingassistantservice.yaml)
```

Comment:
I mean, that's what we're doing here, the vars are in overlays already. Sed is just doing the var substitution.

If we have another overlays explicitly just for variables, then we still have to sed that overlay so we're back at square one. We can't hardcode those values in the overlay because they're dynamic, like the name of the GSA or the Google Cloud project ID.

Unless I'm misunderstanding what you mean?